### DISCUSSION TITLE
why not use blink.cmp's internal lazy loading

### DISCUSSION THREAD

**Discussion by mirsella** [LazyVim/LazyVim#4699] [Q&A] [answered]
hello !

https://github.com/Saghen/blink.cmp says in its example with lazy.nvim to use `lazy = false` as it has its own lazy loading internally.
is there a specific reason for not respecting this and doing it lazy.nvim side ?

just wondering/curious about the inner working

thanks !

**Comment by dotfrag** [comment]
I don't know either, but I suspect `event = "InsertEnter"` is even lazier than what `blink` does.

**Reply by dpetka2001** [reply]
I believe it doesn't matter so much, because it gets loaded from `nvim-lspconfig` (as part of the `capabilities` that are calculated in `lazyvim/plugins/lsp/init.lua`) either way on `LazyFile` (which is an alias for `{ "BufReadPost", "BufNewFile", "BufWritePre" }`) and this already happens before `InsertEnter`. It might as well be `lazy = true` and it would have the same effect. Bottom line is that it just doesn't load as a start-up plugin by `lazy.nvim`.

**Comment by folke** [comment, marked as answer]
That's mainly because `friendly-snippets` is a dependency of blink and that does have some time to load.
It also doesn't really matter since blink works perfectly fine when lazy-loaded on `InsertEnter` or loaded by lspconfig.....
